gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java

```diff
@@ -54,13 +58,14 @@
   protected final int fetchTimeoutMillis;
   protected final int fetchMinBytes;
   protected final int socketTimeoutMillis;
+  protected final KafkaSchemaRegistry schemaRegistry;
 
   public AbstractBaseKafkaConsumerClient(Config config) {
     this.brokers = ConfigUtils.getStringList(config, ConfigurationKeys.KAFKA_BROKERS);
     if (this.brokers.isEmpty()) {
       throw new IllegalArgumentException("Need to specify at least one Kafka broker.");
     }
-
+    this.schemaRegistry = KafkaSchemaRegistry.get(ConfigUtils.configToProperties(config));
```

Comment:
What happens when Kafka Schema Registry is not enabled? Shouldn't we check if the config kafka.schema.registry.url is set?

Reply:
Also - can we do a lazy instantiation of KafkaSchemaRegistry inside getFilteredTopics() method? We only need the schema registry check for filtering the list of topics.

Reply:
1. There is check for schema registry class inside the KafkaSchemaRegistry.get method
  public static <K, S> KafkaSchemaRegistry<K, S> get(Properties props) {
    Preconditions.checkArgument(props.containsKey(KAFKA_SCHEMA_REGISTRY_CLASS),
        "Missing required property " + KAFKA_SCHEMA_REGISTRY_CLASS);
...
}
2. I am assuming we should only initialize it "once" lazily too right?